apps/src/studio/collidable.js

```diff
@@ -104,6 +108,157 @@ Collidable.prototype.bounce = function () {
   }
 };
 
+function getRandomInt(min, max) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+Collidable.prototype.roamGrid = function(type) {
+
+  // Do we have an active location in grid coords?  If not, determine it.
+  if (this.gridX === undefined) {
+    this.gridX = Math.floor(this.x / Studio.SQUARE_SIZE);
+    this.gridY = Math.floor(this.y / Studio.SQUARE_SIZE);
+  }
```

Comment:
Something I like to do to make classes more readable is to take any fields that are going to exist on the object, and initialize them in the constructor. This way I can look at the ctor and get an idea of what all exists on the class. So in this case this would be something like
```
var Collidabale = function () {
  ...
  this.gridX = undefined;
  this.gridY = undefined;
}